add plugin-image-click-response

ChristopheBossens opened the pull request:
### Plugin description
This plugin shows an image on which the user can place points by clicking/touching the image. The location of each point is recorded as data.

### Plugin includes
- [x] Complete code for the plugin
- [x] A readme file following the template
- [x] Documentation in the /docs folder
- [x] An example in the /example folder
- [x] A test suite

### Test case
Available in the examples folder.

jodeleeuw commented:
Hi @ChristopheBossens

Thanks very much for this!

I think it is failing right now because package-lock.json isn't up to date. Can you run `npm i --package-lock-only` in the root directory and then commit the updated package-lock.json file?

ChristopheBossens commented:
Hi @jodeleeuw 

Currently working on another plugin and was wondering if the current one needs anything more for me (want to make sure I am using the correct workflow for working on plugins).

jadeddelta commented on packages/plugin-image-click-response/docs/image-click-response.md:
```diff
@@ -0,0 +1,64 @@
+| dot_radius            | INT    | 5| Radius of the dot in pixels|
+| dot_color             | STRING | lightblue| Color name for the dot|
+| button_label          | STRING | Continue| Label for the continue button |
+| minimum_dots_required | INT    | 0| Minimum number of dots required before the Continue button becomes active|
```
make sure to re-update this value (default -1 in code), also good to explain what -1 does

ChristopheBossens replied:
-1 was a mistake, fixed it in the code

jadeddelta commented on packages/plugin-image-click-response/src/index.ts:
```diff
@@ -0,0 +1,179 @@
+      type: ParameterType.INT,
+    },
+    /** Array of (x,y) values for each point in the image */
+    points: {
```
wonder if it would also be wise to collect the time of when each point was clicked?

ChristopheBossens replied:
Great idea! That is now included!

jadeddelta commented on packages/plugin-image-click-response/src/index.ts:
```diff
@@ -0,0 +1,179 @@
+    // Create layout
+    let html = "";
+    let preamble_html = `<div id="jspsych-survey-text-preamble" class="jspsych-survey-text-preamble">${trial.preamble}</div>`;
+    let button_html = `<div><button id='image-click-response-button'>${trial.button_label}</button></div>`;
```
don't forget to add the `jspsych-btn` class to buttons to keep in line with jsPsych styling

jadeddelta commented on packages/plugin-image-click-response/src/index.ts:
```diff
@@ -0,0 +1,179 @@
+
+    // Create layout
+    let html = "";
+    let preamble_html = `<div id="jspsych-survey-text-preamble" class="jspsych-survey-text-preamble">${trial.preamble}</div>`;
```
wrong plugin? the css will also only work if you have the jspsych-survey-text plugin css as well, so would be nice to either remove or if you really need the css, add a css tag inside

ChristopheBossens replied:
Again well spotted, thanks!

jadeddelta commented on packages/plugin-image-click-response/src/index.ts:
```diff
@@ -0,0 +1,179 @@
+    let html = "";
+    let preamble_html = `<div id="jspsych-survey-text-preamble" class="jspsych-survey-text-preamble">${trial.preamble}</div>`;
+    let button_html = `<div><button id='image-click-response-button'>${trial.button_label}</button></div>`;
+    let svg_html = `<div><svg id='image-click-response-svg'><image id='image-click-response-image' href='${trial.stimulus}'></svg></div>`;
```
this is the primary one that i'm worried about- this works perfectly fine on chrome! but on firefox, the image refuses to learn. it might be a combination of various factors, but from what i was able to gather from some basic testing, it helped to add the namespace declarations to the svg element (`xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"`), along with manually setting the height/width of the svg to accommodate for firefox being stricter with image loads in svgs might work well. i haven't found a fix at the moment that satisfyingly accommodates for different shapes of images, these are just my heads up if you want to have firefox compatible as well- or if you just want to make it chrome only and make a note to developers to use exclusion criteria against firefox.

ChristopheBossens replied:
Apparently I was setting the size of the svg container in the image.onload function, but forgot to include the "px" units. Chrome was more forgiving about this, but it works on Firefox as well now :)

jadeddelta reviewed (changes requested):
hi @ChristopheBossens, i've been asked to take this on and i've left some basic feedback-- all the development protocols that you've been working through are fantastic so far.. just have two tips: 

1. make sure to `npm i` frequently, especially after merges so that the package-lock.json can regenerate, if the package-lock.json isn't up to date and is missing dependencies it will fail npm's clean install which depends on the package-lock.json

2. be sure to run `npm run changeset` in the repo's root directory to add a changeset! this helps us with versioning a ton

ChristopheBossens commented:
Hi jadeddelta,

Thank you for reviewing the code and your comments! I have addressed each of them, added a changeset and pushed my changes again. Let me know if there is anything else I need to do (first time I am following this workflow on Github so not 100% sure if I am doing everything correctly).

jadeddelta commented:
hey @ChristopheBossens, almost everything looks good to go! just two more things.

1. make sure to update the README.md to indicate that this plugin is only compatible with jsPsych 8.0+
2. wondering about you making the changeset minor or major: if you set a newly released (package.json version is 0.x.x) plugin to a minor bump, it means the plugin is basically in alpha/beta and will change. if you're finished with the plugin, change it to a major bump so that it will start out at 1.0.0.

ChristopheBossens commented:
Updated the compatibility and changed the changeset to a major release :)

jadeddelta reviewed (approved):
it all looks good to me! i'll go ahead and merge these changes and we'll release this with another PR coming up soon!
